plugins/weida-raw/fu-weida-raw-device.c

```diff
@@ -0,0 +1,1093 @@
+/*
+ * Copyright 2024 Randy Lai <randy.lai@weidahitech.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "config.h"
+
+#include <fcntl.h>
+#include <linux/hidraw.h>
+#include <linux/input.h>
+
+#include "fu-weida-raw-common.h"
+#include "fu-weida-raw-device.h"
+#include "fu-weida-raw-struct.h"
+
+/* this can be set using Flags=example in the quirk file  */
+#define FU_WEIDA_RAW_DEVICE_FLAG_EXAMPLE (1 << 0)
+
+#define FU_WEIDA_RAW_DEVICE_IOCTL_TIMEOUT 5000 /* ms */
+#define FU_WEIDA_RAW_REQ_DEV_INFO	  0xF2
+#define FU_WEIDA_RAW_RETRY_COUNT	  5
+#define FU_WEIDA_RAW_PAGE_SIZE		  0x1000
+#define FU_WEIDA_RAW_FLASH_PAGE_SIZE	  256
+#define FU_WEIDA_RAW_USB_MAX_PAYLOAD_SIZE 63
+
+#define FU_WEIDA_RAW_FOURCC_ID_RIFF 0x46464952
+#define FU_WEIDA_RAW_FOURCC_ID_WIF2 0x32464957
+#define FU_WEIDA_RAW_FOURCC_ID_WHIF 0x46494857
+#define FU_WEIDA_RAW_FOURCC_ID_INFO 0x4f464e49
+#define FU_WEIDA_RAW_FOURCC_ID_FSUM 0x4d555346
+#define FU_WEIDA_RAW_FOURCC_ID_FERA 0x41524546
+#define FU_WEIDA_RAW_FOURCC_ID_FBIN 0x4e494246
+#define FU_WEIDA_RAW_FOURCC_ID_FRMT 0x544D5246
+#define FU_WEIDA_RAW_FOURCC_ID_FRWR 0x52575246
+#define FU_WEIDA_RAW_FOURCC_ID_CNFG 0x47464E43
+
+typedef struct {
+	guint32 FourCC;
+	guint32 FileSize;
+	guint32 DataType;
```

Comment:
struct fields need to be lower case please

Reply:
Done